### DISCUSSION TITLE
How to Send Authenticated Requests in React Native WebView

### DISCUSSION THREAD

**Discussion by RafaelCENG** [supabase/supabase#17694] [Questions]
Hello everyone, I'm working on integrating my web dashboard into a React Native WebView. I've successfully displayed my web app, but I'm encountering an issue when trying to navigate beyond the sign-in page. I have the user's access_token for authentication. Can you provide guidance on how to send authenticated requests to access different parts of my web app within the WebView?
 ```js
 <WebView
      source={{ uri: "url/dashboard" }}
      style={styles.webView}
      originWhitelist={["*"]}
      nestedScrollEnabled
      //injectedJavaScript={injectJavaScriptCode}
    />
```
    
    
[Working with custom headers, sessions, and cookies](https://github.com/react-native-webview/react-native-webview/blob/4c050771757f4f7d92cf816bf0a5bf16c5539c07/docs/Guide.md#working-with-custom-headers-sessions-and-cookies)
    
There is a small guide on react-native-webview but still couldn't make it.

**Comment by toy-crane** [comment]
hi! do you find the way to solve?

**Reply by RafaelCENG** [reply]
Hi my senior did by injecting javascript using local-storage.

**Comment by damien-schneider** [comment]
Any update on this? I would love to have a solution, or at least to use the react-native supabase auth flow but then passing the auth token to the webview to have for example a Nextjs app running without having 2 codebases to maintain.
